lib/XTR/StatsMigrationTask.php

```diff
@@ -0,0 +1,481 @@
+
+        // read configOverride
+        $configOverrideFile = $this->getOption('configOverride', '');
+        if (file_exists($configOverrideFile)) {
+
+            $contents = json_decode(file_get_contents($configOverrideFile), true);
+
+            $options = [
+                'killSwitch' => $contents['killSwitch'],
+                'numberOfRecords' => $contents['numberOfRecords'],
+                'numberOfLoops' => $contents['numberOfLoops'],
+                'pauseBetweenLoops' => $contents['pauseBetweenLoops'],
+                'optimiseOnComplete' => $contents['optimiseOnComplete'],
+            ];
+
+        } else {
+
+            // Config options
+            $options = [
+                'killSwitch' => $this->getOption('killSwitch', 0),
+                'numberOfRecords' => $this->getOption('numberOfRecords', 10000),
+                'numberOfLoops' => $this->getOption('numberOfLoops', 1000),
+                'pauseBetweenLoops' => $this->getOption('pauseBetweenLoops', 10),
+                'optimiseOnComplete' => $this->getOption('optimiseOnComplete', 1),
+            ];
+        }
+
+        if ($options['killSwitch'] == 1) {
+
+            // Check stat_archive table exists
+            $this->archiveExist = $this->store->exists('SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_SCHEMA = :schema AND TABLE_NAME = :name', [
+                'schema' => $_SERVER['MYSQL_DATABASE'],
+                'name' => 'stat_archive'
+            ]);
+
+            // Get timestore engine
+            $timeSeriesStore = $this->timeSeriesStore->getEngine();
+
+            if ($timeSeriesStore == 'mongodb') {
+                $this->moveStatsFromStatArchiveToStatMongoDb($options);
+            }
+
+            // If when the task runs it finds that MongoDB is disabled,
+            // and there isn't a stat_archive table, then it should disable itself and not run again
+            // (work is considered to be done at that point).
+            else {
+
+                if ($this->archiveExist == true) {
+                    $this->moveStatsFromStatArchiveToStatMysql($options);
+
+                } else {
+                    // Disable the task
+                    $this->getTask()->isActive = 0;
+                    $this->getTask()->save();
+                }
+            }
+        }
+    }
+
+    public function moveStatsFromStatArchiveToStatMysql($options = [])
+    {
+
+        $fileName = $this->config->getSetting('LIBRARY_LOCATION') . '.watermark_stat_archive_mysql.txt';
+
+        $options = array_merge([
+            'numberOfRecords' => 10000,
+            'numberOfLoops' => 1000,
+            'pauseBetweenLoops' => 10,
+        ], $options);
+
+        // Get low watermark from file
+        $watermark = $this->getWatermarkFromFile($fileName, 'stat_archive');
+
+        $numberOfLoops = 0;
+        $count = 0;
+        while ($watermark > 0) {
```

Comment:
will the watermark ever be 0?

Reply:
no it will not. watermark is 1 if there are no records in stat/stat_archive.

Reply:
so that means this loop wont ever finish?

the lowest statId in the table might be 8938549382 - do we handle that?

Reply:
we do make it -1 when end of records is reached. will look into it in the morning again.

Reply:
in checkEndOfRecords()